src/StackExchange.Redis/ServerEndPoint.cs

```diff
@@ -935,10 +935,14 @@ private async Task HandshakeAsync(PhysicalConnection connection, LogProxy? log)
                     // server version, so we will use this speculatively and hope for the best
                     log?.WriteLine($"{Format.ToString(this)}: Setting client lib/ver");
```

Comment:
a bit late now, but: maybe this should be inside the null/empty check

Reply:
Agree it's weird, but there are 2 distinct checks and 1 log message atm - we could log each, change the method, or something else - thoughts? Happy to tidy up!